appveyor.yml

```diff
@@ -1,48 +1,48 @@
 environment:
-  COVERALLS_REPO_TOKEN:
-    secure: 2NJ5Ct55cHJ9WEg3xbSqCuv0rdgzzb6pnzOIG5OkMbTndw3wOBrXntWFoQrXiMFi
-    # this is pytest's token in coveralls.io, encrypted
-    # using pytestbot account as detailed here:
-    # https://www.appveyor.com/docs/build-configuration#secure-variables
-
   matrix:
-  # coveralls is not in the default env list
-  - TOXENV: "coveralls"
-  # note: please use "tox --listenvs" to populate the build matrix below
   - TOXENV: "linting"
+    PYTEST_NO_COVERAGE: "1"
   - TOXENV: "py27"
   - TOXENV: "py34"
   - TOXENV: "py35"
   - TOXENV: "py36"
   - TOXENV: "py37"
   - TOXENV: "pypy"
-  - TOXENV: "py27-pexpect"
+    PYTEST_NO_COVERAGE: "1"
   - TOXENV: "py27-xdist"
   - TOXENV: "py27-trial"
   - TOXENV: "py27-numpy"
   - TOXENV: "py27-pluggymaster"
-  - TOXENV: "py36-pexpect"
   - TOXENV: "py36-xdist"
   - TOXENV: "py36-trial"
   - TOXENV: "py36-numpy"
   - TOXENV: "py36-pluggymaster"
   - TOXENV: "py27-nobyte"
   - TOXENV: "doctesting"
   - TOXENV: "py36-freeze"
+    PYTEST_NO_COVERAGE: "1"
   - TOXENV: "docs"
+    PYTEST_NO_COVERAGE: "1"
 
 install:
   - echo Installed Pythons
   - dir c:\Python*
 
   - if "%TOXENV%" == "pypy" call scripts\install-pypy.bat
 
+  - C:\Python36\python -m pip install --upgrade pip
   - C:\Python36\python -m pip install --upgrade --pre tox
 
 build: false  # Not a C# project, build stuff at the test step instead.
 
+before_test:
+  - call scripts\prepare-coverage.bat
```

Comment:
I believe this does not export the necessary env vars?
Should rather be done inline here.

Reply:
> I believe this does not export the necessary env vars?

You mean if the env vars exported by the script will be available in the environment for the other commands? They will be according to [the docs](https://www.appveyor.com/docs/appveyor-yml/):

```yaml
# scripts to run before tests (working directory and environment changes are persisted from the previous steps such as "before_build")
before_test:
  - echo script1
  - ps: Write-Host "script1"
```

Also we can see that's true because for example `coverage-enable-subprocess` gets installed by tox:

```
py27 installed: atomicwrites==1.2.1,attrs==18.2.0,certifi==2018.8.24,chardet==3.0.4,colorama==0.3.9,coverage==4.5.1,coverage-enable-subprocess==1.0,enum34==1.1.6,funcsigs==1.0.2,hypothesis==3.70.0,idna==2.7,mock==2.0.0,more-itertools==4.3.0,nose==1.3.7,pathlib2==2.3.2,pbr==4.2.0,pluggy==0.7.1,py==1.6.0,pytest==3.7.5.dev40+g759ee5a2,requests==2.19.1,scandir==1.9.0,six==1.11.0,urllib3==1.23
```

Reply:
Yes, noticed this myself.  Thought about UNIX shell scripts, where you would have to source them.